mcs/class/System.Security/Mono.Security.Cryptography/ManagedProtection.cs

```diff
@@ -237,23 +237,29 @@ public static byte[] Unprotect (byte[] encryptedData, byte[] optionalEntropy, Da
 
 		private static RSA user;
 		private static RSA machine;
+		private readonly static object user_lock = new object ();
+		private readonly static object machine_lock = new object ();
 
 		private static RSA GetKey (DataProtectionScope scope)
 		{
 			switch (scope) {
 			case DataProtectionScope.CurrentUser:
 				if (user == null) {
-					CspParameters csp = new CspParameters ();
-					csp.KeyContainerName = "DAPI";
-					user = new RSACryptoServiceProvider (1536, csp);
+					lock (user_lock) {
```

Comment:
Yes, but it doesn't really matter in this case since in the worst case we just create another object with the same parameters. I can add it if you prefer following the double-check pattern